### ISSUE TITLE
Crashplan reindexes all the files and lost the files history after upgrading

### ISSUE THREAD

**Issue by nacesprin** [jlesage/docker-crashplan-pro#379]
Is it normal that following the guide in https://github.com/jlesage/docker-crashplan#synology crashplan reupload to crashplan servers all the files in NAS like a fresh installation? I must say that host volumes are keeped, but a new anonymous volume is created in the docker container recreation.
So, I dont know if I must to reuse the old anonymous volume (approx.500Gb) in the new created container after the container upgrade, or I am really not doing the upgrade proccess correctly.

Thanks

**Comment by nacesprin** [comment]
I have many volumes mapped and works fine, but after the image upgrade to the "latest" tag, I realize by entering in crashplan-pro webapplication, the backup begin to count up from 0 bytes until ~500Gb and the files history is not accessible.
I discovered browsing the /volume1/@docker/volumes directory there was 2 big sized directories, each of the are ~500Gb approx: one of the is used for the current image. I suspect the other big sized directory (also 500Gb) was of the previous crashplan-pro image.
I dont know if in this case, I will need to reuse the same volume when upgrading to the next version of crashplan-pro.

Am I doing something wrong?

**Comment by jlesage** [comment]
> I have many volumes mapped and works fine

Ok but do you have one for `/config` ?

After the upgrade, did you have to setup CrashPlan like if it was a new device ?

Can you share the configuration of the container ?

**Comment by nacesprin** [comment]
> After the upgrade, did you have to setup CrashPlan like if it was a new device ?

No. Crashplan web app and config remains untouched. But instead of resume the files backup on the last point was stopped, it begins from scratch with a fresh backup files and it remains during 48h backing up  all the mapped directories, approx ~500Gb.
And thus I think for the next time crashplan need to be updated, I will reuse the unusued volume instead of the new volume which is created with the new image. But I dont know if I am acting wrong.

### volumes and container configuration:
```
{
   "CapAdd" : null,
   "CapDrop" : null,
   "cmd" : "/init",
   "cpu_priority" : 50,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : true,
   "enabled" : true,
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "LANG",
         "value" : "en_US.UTF-8"
      },
      {
         "key" : "S6_BEHAVIOUR_IF_STAGE2_FAILS",
         "value" : "3"
      },
      {
         "key" : "S6_SERVICE_DEPS",
         "value" : "1"
      },
      {
         "key" : "USER_ID",
         "value" : "0"
      },
      {
         "key" : "GROUP_ID",
         "value" : "0"
      },
      {
         "key" : "APP_NAME",
         "value" : "CrashPlan for Small Business"
      },
      {
         "key" : "APP_USER",
         "value" : "app"
      },
      {
         "key" : "XDG_DATA_HOME",
         "value" : "/config/xdg/data"
      },
      {
         "key" : "XDG_CONFIG_HOME",
         "value" : "/config/xdg/config"
      },
      {
         "key" : "XDG_CACHE_HOME",
         "value" : "/config/xdg/cache"
      },
      {
         "key" : "XDG_RUNTIME_DIR",
         "value" : "/tmp/run/user/app"
      },
      {
         "key" : "DISPLAY",
         "value" : ":0"
      },
      {
         "key" : "DISPLAY_WIDTH",
         "value" : "1280"
      },
      {
         "key" : "DISPLAY_HEIGHT",
         "value" : "768"
      },
      {
         "key" : "S6_WAIT_FOR_SERVICE_MAXTIME",
         "value" : "10000"
      },
      {
         "key" : "KEEP_APP_RUNNING",
         "value" : "1"
      },
      {
         "key" : "TZ",
         "value" : "Europe/Madrid"
      }
   ],
   "exporting" : false,
   "id" : "d94e7451e150cc855c5c9150a4f86e3b728eb9753d32d0f2bbba661b9e138911",
   "image" : "jlesage/crashplan-pro:latest",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "jlesage-crashplan-pro",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "bridge",
   "port_bindings" : [
      {
         "container_port" : 5800,
         "host_port" : 5800,
         "type" : "tcp"
      }
   ],
   "privileged" : false,
   "shortcut" : {
      "enable_shortcut" : false,
      "enable_status_page" : false,
      "enable_web_page" : false,
      "web_page_url" : ""
   },
   "use_host_network" : false,
   "volume_bindings" : [
      {
         "host_volume_file" : "/music",
         "mount_point" : "/storage/music",
         "type" : "ro"
      },
      {
         "host_volume_file" : "/photo",
         "mount_point" : "/storage/photo",
         "type" : "ro"
      },
      {
         "host_volume_file" : "/SynologyDriveJavier",
         "mount_point" : "/storage/SynologyDriveJavier",
         "type" : "ro"
      },
      {
         "host_volume_file" : "/video",
         "mount_point" : "/storage/video",
         "type" : "ro"
      },
      {
         "host_volume_file" : "/Autonomo",
         "mount_point" : "/storage/Autonomo",
         "type" : "ro"
      },
      {
         "host_volume_file" : "/EscanerLaser",
         "mount_point" : "/storage/EscanerLaser",
         "type" : "ro"
      },
      {
         "host_volume_file" : "/docker/appdata/crashplan-pro",
         "mount_point" : "/config",
         "type" : "rw"
      }
   ]
}

```

Thanks for your support

**Comment by jlesage** [comment]
> No. Crashplan web app and config remains untouched. But instead of resume the files backup on the last point was stopped, it begins from scratch with a fresh backup files and it remains during 48h backing up all the mapped directories, approx ~500Gb.

So after an update, it's normal for CrashPlan to re-synchronize the data.  At some point it might look like if everything is re-uploaded, but if you check at Tools->Historty, you should see that files are "uploaded" at a very high "effective" rate.  This in fact shows that data is already in cloud and nothing is really transferred.

> And thus I think for the next time crashplan need to be updated, I will reuse the unusued volume instead of the new volume which is created with the new image. But I dont know if I am acting wrong.

Not sure what this volume would be.  Everything under `volume_bindings` looks good.

**Comment by DraGula** [comment]
Hello there,

I have the same problem since the last update :
CrashPlan v10.4.0 (docker image v22.11.1)

Crashplan has completed the initialization. Backup set is at 0 bytes
In Manage I can browse my "storage" directory, I see contents and the size indicated is correct.

I do not understand
Can I check something?
Thank you!

**Comment by DraGula** [comment]
> Hello there,
> 
> I have the same problem since the last update : CrashPlan v10.4.0 (docker image v22.11.1)
> 
> Crashplan has completed the initialization. Backup set is at 0 bytes In Manage I can browse my "storage" directory, I see contents and the size indicated is correct.
> 
> In history : Scanning for files completed in < 1 minute: 0 files (0MB) found ?!
> 
> I do not understand Can I check something? Thank you!

Forget it...
I just forced a "backup.scan".
It seems ok. Strange!

**Comment by idaband** [comment]
I'm running a synology and missed the part of using a config directory.  

A) Whats the correct volume to use for mapping volumes?   /docker/appdata/crashplan-pro:/config
Do i need to setup anything special permission wise on the synology for that path?

B) is it too late to fix this?  I don't want to upload 3 months of data again
